### PR TITLE
Integrated Google Calendar on the Events page

### LINKED ISSUE #121: coding: integrate google calendar in events page
## Description
As a developer, I need to integrate google calendar in events page , so that the user can see all the events related to community.

## Mocks
image

the spacing is 16 from both sides

### Update [Required]
- [ ]  Follows all the design guidelines as mentioned and is responsive enough on large screen devices.
- [ ]  the events which we organise should be updated in the calendar as mentioned

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by shravani05**
### Description

Integrated Google Calendar on the Events page. 

Fixes #121 

### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

image

This can be tested here https://shravani05.github.io/anitab-org.github.io/

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have attached link of deployed site. https://shravani05.github.io/anitab-org.github.io/
- [ ] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by Bucky25** [comment]
@shravani05 please add a link of deployed gh-page

**Comment by shravani05** [comment]
> @shravani05 please add a link of deployed gh-page

@Bucky25  Here is the deployed link: https://shravani05.github.io/anitab-org.github.io/
I have also attached the image of the Events page
Cal

**Comment by tichnas** [comment]
@shravani05 Thanks for this PR :)
Also, it would be great if you change the inline styles to react-native Stylesheet

**Comment by shravani05** [comment]
> @shravani05 Thanks for this PR :)
> Also, it would be great if you change the inline styles to react-native Stylesheet

@tichnas I have changed the styles to react-native StyleSheet for all the components. But the iframe component can't receive style created by StyleSheet.create(), so I have kept inline style for it.

File: src/Components/Events/GoogleCalendar/index.js [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
-export default GoogleCalendar;
\ No newline at end of file
+const styles = StyleSheet.create({
+  container: {
+    width: "80%",
+    alignItems: 'left',
+    paddingLeft: 16,
+    paddingRight: 16,
+  }
+});
```
Comment (Bucky25) [inline src/Components/Events/GoogleCalendar/index.js:22]: @shravani05 please add some padding between above para and calendar heading

File: src/Components/Events/index.js [hunk 2]
```diff
@@ -1,23 +1,12 @@
...
-      }}
-    >
-    <Text h1 style={{
-          fontSize: "30px",
-          fontWeight: "bold",
-          color: "#013287"
-      }}>EVENT CALENDAR</Text> 
-      <Text style={{color: "#013287"}}>
+    <View style={styles.container}>
+      <Text h1 style={styles.heading}>EVENT CALENDAR</Text>
```
Comment (Bucky25) [inline src/Components/Events/index.js:8]: please sectionsubheader to write sub heading

**Review by Bucky25 — CHANGES_REQUESTED** [review]
please use design give to add padding or margin between components

**Comment by shravani05** [comment]
@Bucky25 please let me know if there are any other changes to be made 
https://shravani05.github.io/anitab-org.github.io/
Cal1

File: src/Components/Events/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  return (  
+      <View style={styles.container}>
+      <SectionSubheader title="Event Calendar"/>
+      <Text style={styles.data}>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer malesuada massa nec porta interdum. 
+      Vivamus porta mollis urna at congue. 
+      Maecenas mollis, dolor ut auctor ultrices, diam ex dignissim ante, at sollicitudin risus risus id felis. 
+      Suspendisse vel ante placerat, feugiat ligula pretium, dignissim est. 
+      Donec vestibulum ullamcorper sapien et egestas. Ut eu aliquet leo.
+      </Text>
```
Comment (Bucky25) [inline src/Components/Events/index.js:16]: @shravani05  please style this text like other pages having text below heading

**Review by Bucky25 — CHANGES_REQUESTED** [review]

**Comment by shravani05** [comment]
@Bucky25 I have added styles as per other pages. Please let me know if any further changes are required.

File: src/Components/Events/GoogleCalendar/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { MainContainer } from '../style';
+
+function GoogleCalendar() {
+  return (
+    <MainContainer>
+        {/* Replace this iframe with the iframe of the google calendar associated with the google account according to which you want the schedule */}
+        <iframe src="https://calendar.google.com/calendar/embed?src=er3gi62a5hip5jnk1f7r0al62s%40group.calendar.google.com&ctz=America%2FLos_Angeles"
```
Comment (isabelcosta) [inline src/Components/Events/GoogleCalendar/index.js:8]:
```suggestion
        <iframe src="https://calendar.google.com/calendar/embed?src=er3gi62a5hip5jnk1f7r0al62s%40group.calendar.google.com" 
```
I think if we don't put the `ctz` query parameter, the  timezone assumed will be the one from the computer accessing the page.

File: src/Components/Events/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+function Events() {
+
+  const renderCalendar = () => {
+    return(
+      <Box>
+        <Content>
+          <SectionSubheader title="Event Calendar"/>
+          <Description>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer malesuada massa nec porta interdum.
```
Comment (isabelcosta) [inline src/Components/Events/index.js:14]: Not sure about the content yet, but you could mention the external links to the calendar in different timezones. Here's the text you can put:
```suggestion
          We host multiple open sessions to the community where members of the community can discuss concerns about the community, ask questions, discuss projects, etc. These sessions are open to anyone and are usually conducted via Zoom. These are audio-only calls, so no video is required.
          You can see our calendar in different timezones:
```

Then in front of ":" you could add HTML code to link each timezone to its calendar like: ([PST](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=America/Los_Angeles) || [EST](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=America/New_York) || [GMT](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=GMT) || [WAT](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=Africa/Lagos) || [IST](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=Asia/Colombo) || [AEDT](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=Australia/Sydney))

@shravani05

**Review by isabelcosta — CHANGES_REQUESTED** [review]
This is looking amazing, I would like to suggest a small improvement. Please read my comment about the calendar in different timezones, and let me know if you need any clarification :) @shravani05 

Thank you so much for your contribution!

File: src/Components/Events/index.js (hunk 5 above)
Comment (shravani05) [reply src/Components/Events/index.js:14]: @isabelcosta On it!

**Comment by shravani05** [comment]
@isabelcosta I have added the links as mentioned. Please let me know if any further changes are required :)

**Review by isabelcosta — APPROVED** [review]
I tested this with the link you provided, and functionality wise this looks amazing 💯 🎉 I learned a little bit from your implementation :D

Great work @shravani05 👏 👏 👏 
Thank you so much for this contribution!

**Comment by isabelcosta** [comment]
@shravani05 as a suggestion for your future contributions, if you do a change to user interface, add an image of the UI to the PR description and also add the link you sent on the latest comment. I just did that for you in your PR description, now it looks much more informative for whoever reviews your work :)

**Comment by mayburgos** [comment]
@anitab-org/admins Quick note: Add another issue to see if we can research how to rename through the URL link.

**Review by tichnas — CHANGES_REQUESTED** [review]
Delete package-lock.json file as it isn't required for yarn package manager (which we are using).
Also, instead of using `var`, use `const` or `let`.

File: src/Components/Events/style.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { View, Text } from 'react-native';
```
Comment (tichnas) [inline src/Components/Events/style.js:1]: Correct the indentation in this file

File: src/Components/Events/index.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    )
+  });
+
+  const renderCalendar = () => {
+    return(
+      <Box>
+        <Content>
+          <SectionSubheader title="Event Calendar"/>
+          <Description>
+          We host multiple open sessions to the community where members of the community can discuss concerns about the community, ask questions, discuss projects, etc. These sessions are open to anyone and are usually conducted via Zoom. These are audio-only calls, so no video is required.
```
Comment (tichnas) [inline src/Components/Events/index.js:60]: Not sure, but if this paragraph isn't constant we should put it somewhere outside the code (inside content folder)
What do you say @isabelcosta?

**Comment by shravani05** [comment]
> @shravani05 as a suggestion for your future contributions, if you do a change to user interface, add an image of the UI to the PR description and also add the link you sent on the latest comment. I just did that for you in your PR description, now it looks much more informative for whoever reviews your work :)

Thank you so much for the suggestion😊@isabelcosta 
It has been great working on this issue!

File: src/Components/Events/GoogleCalendar/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { MainContainer } from '../style';
+
+function GoogleCalendar({timezone}) {
+  return (
+    <MainContainer>
+        <iframe src={`https://calendar.google.com/calendar/u/0/embed?src=sh10tv3mtfve62somg9nngp9tg@group.calendar.google.com${timezone}`}
+          title="Calendar" style={{border: "0", width: "100%", height:"500px", frameborder:"0", scrolling:"no"}}>
```
Comment (annabauza) [inline src/Components/Events/GoogleCalendar/index.js:8]: can we move style from inline?

File: src/Components/Events/index.js (hunk 7 above)
Comment (annabauza) [inline src/Components/Events/index.js:60]: Can we move this to content, please?

File: src/Components/Events/GoogleCalendar/index.js (hunk 8 above)
Comment (annabauza) [reply src/Components/Events/GoogleCalendar/index.js:8]: are pixels necessary ?

**Review by annabauza — CHANGES_REQUESTED** [review]
few tweaks and it looks really good. Thank you!

File: src/Components/Events/index.js (hunk 7 above)
Comment (isabelcosta) [reply src/Components/Events/index.js:60]: agreed. I was not aware of the whole code architecture, that seems to make sense @tichnas :)

File: src/Components/Events/GoogleCalendar/index.js (hunk 8 above)
Comment (shravani05) [reply src/Components/Events/GoogleCalendar/index.js:8]: @annabauza  I tried adding react-native StyleSheet for the iframe, but the iframe component can't receive style created by StyleSheet.create(), so I have kept inline style for it. Please let me know if there is any other way to do it.
And pixels are necessary to get the required height.

**Comment by shravani05** [comment]
Sorry for delay in the further commits, I was busy last month with my exams. @tichnas I have replaced the var with const and also corrected the indentation in the style.js file. Also, I have moved the description to the content folder. Please let me know if any further changes are required :)
The deployed site can be found here: https://shravani05.github.io/anitab-org.github.io/

File: src/Components/Events/GoogleCalendar/index.js (hunk 8 above)
Comment (annabauza) [reply src/Components/Events/GoogleCalendar/index.js:8]: Fair enough

**Review by annabauza — APPROVED** [review]
:100:

**Comment by isabelcosta** [comment]
@shravani05 can you please fix the merge conflicts?

**Comment by shravani05** [comment]
@isabelcosta I have resolved the merge conflicts. Please have a look here:  https://shravani05.github.io/anitab-org.github.io/

**Comment by isabelcosta** [comment]
@annabauza this has 2 approvals, so it's ready for merging. Is this correct?
I checked it on @shravani05 website https://shravani05.github.io/anitab-org.github.io/ and it seems to be OK 👏 

image

**Pull request merged**